### PR TITLE
fix(ru): article Equality_comparisons_and_sameness#same-value_equality_using_object.is

### PR DESCRIPTION
**Pull request by scffs**
I tried to translate the text into Russian

### CODE REVIEW

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 1]
```diff
@@ -186,23 +186,23 @@ function attemptMutation(v) {
...
+Нулевое значение с одинаковым значением отличается от строгого равенства только тем, что рассматривает `NaN` как эквивалент, и отличается от равенства с одинаковым значением только тем, что рассматривает `-0` как эквивалент `0`. Это обычно приводит к наиболее разумному поведению во время поиска, особенно при работе с `NaN`. Он используется с помощью [`Array.prototype.includes()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), [`TypedArray.prototype.includes()`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/includes), также как [`Map`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) и [`Set`](/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) методы сравнения равенства ключей.
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:205]:
```suggestion
_Равенство одинаковых величин и нулей_ отличается от _строгого равенства_ тем, что принимает каждое значение `NaN` равным любому другому значению `NaN`, а от _равенства одинаковых величин_ тем, что принимает `-0` равным `0`. Подобное поведение обычно оказывается самым уместным при поиске в списках, особенно при работе с `NaN`. Данная стратегия сравнения используется в методах [`Array.prototype.includes()`](/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/includes), [`TypedArray.prototype.includes()`](/ru/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/includes), а так же в [`Map`](/ru/docs/Web/JavaScript/Reference/Global_Objects/Map) и [`Set`](/ru/docs/Web/JavaScript/Reference/Global_Objects/Set) для оценки равенства ключей.
```

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 2]
```diff
@@ -243,21 +243,21 @@ Same-value-zero only differs from strict equality by treating `NaN` as equivalen
 | `'foo'`             | `NaN`               | `❌ false` | `❌ false` | `❌ false`  | `❌ false`      |
 | `NaN`               | `NaN`               | `❌ false` | `❌ false` | `✅ true`   | `✅ true`       |
 
-### When to use Object.is() versus triple equals
+### Когда использовать Object.is (), а когда ===
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:246]:
```suggestion
### Когда использовать Object.is(), а когда ===
```

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 3]
```diff
@@ -243,21 +243,21 @@ Same-value-zero only differs from strict equality by treating `NaN` as equivalen
...
+В общем, единственный раз, когда {{jsxref("Object.is")}} особое поведение по отношению к нулям, вероятно, будет представлять интерес при реализации определенных схем метапрограммирования, особенно в отношении дескрипторов свойств, когда желательно, чтобы ваша работа отражала некоторые характеристики {{jsxref("Object.defineProperty")}}. Если ваш вариант использования этого не требует, рекомендуется избегать {{jsxref("Object.is")}} и вместо этого используйте [`===`](/en-US/docs/Web/JavaScript/Reference/Operators). Даже если ваши требования предполагают сравнение между двумя значениями {{jsxref("NaN")}}, равными `true`, как правило, проще выполнить проверку {{jsxref("NaN")}} в специальном случае (используя метод {{jsxref("isNaN")}} доступно в предыдущих версиях ECMAScript), чем выяснить, как окружающие вычисления могут повлиять на знак любых нулей, с которыми вы сталкиваетесь при сравнении.
 
-Here's a non-exhaustive list of built-in methods and operators that might cause a distinction between `-0` and `+0` to manifest itself in your code:
+Вот неполный список встроенных методов и операторов, которые могут привести к тому, что различие между `-0` и `+0` проявится в вашем коде:
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:250]:
```suggestion
Вот неполный список встроенных методов и операторов, при использовании которых, различие между `-0` и `+0` может оказаться важным для вашего кода:
```

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 4]
```diff
@@ -243,21 +243,21 @@ Same-value-zero only differs from strict equality by treating `NaN` as equivalen
...
 
-  - : Consider the following example:
+  - : Рассмотрим следующий пример:
 
     ```js
     const stoppingForce = obj.mass * -obj.velocity;
     ```
 
-    If `obj.velocity` is `0` (or computes to `0`), a `-0` is introduced at that place and propagates out into `stoppingForce`.
+    Если `obj.velocity` равно `0` (или вычисляется до `0`), в этом месте вводится `-0` и распространяется в `stoppingForce`.
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:260]:
```suggestion
    Если `obj.velocity` равно `0` (или вычисляется как `0`), в этом месте возникает `-0` и присваивается в `stoppingForce`.
```

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 5]
```diff
@@ -243,21 +243,21 @@ Same-value-zero only differs from strict equality by treating `NaN` as equivalen
...
+В общем, единственный раз, когда {{jsxref("Object.is")}} особое поведение по отношению к нулям, вероятно, будет представлять интерес при реализации определенных схем метапрограммирования, особенно в отношении дескрипторов свойств, когда желательно, чтобы ваша работа отражала некоторые характеристики {{jsxref("Object.defineProperty")}}. Если ваш вариант использования этого не требует, рекомендуется избегать {{jsxref("Object.is")}} и вместо этого используйте [`===`](/en-US/docs/Web/JavaScript/Reference/Operators). Даже если ваши требования предполагают сравнение между двумя значениями {{jsxref("NaN")}}, равными `true`, как правило, проще выполнить проверку {{jsxref("NaN")}} в специальном случае (используя метод {{jsxref("isNaN")}} доступно в предыдущих версиях ECMAScript), чем выяснить, как окружающие вычисления могут повлиять на знак любых нулей, с которыми вы сталкиваетесь при сравнении.
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:248]:
```suggestion
Особое поведение {{jsxref("Object.is")}} по отношению к нулям, вероятно, будет представлять интерес при реализации определенных схем метапрограммирования, особенно в отношении дескрипторов свойств, когда желательно, чтобы ваш алгоритм имел такое же поведение, как {{jsxref("Object.defineProperty")}}. В случае, если вам этого не требуется, рекомендуется избегать {{jsxref("Object.is")}} и вместо этого использовать [`===`](/ru/docs/Web/JavaScript/Reference/Operators). Как правило, даже при необходимости сравнения {{jsxref("NaN")}}-значений, эффективнее выполнить проверку таких значений отдельно, при помощи метода {{jsxref("isNaN")}}, чем выяснять, как окружающие вычисления могут повлиять на знаки нулей и как это отразится на вашем алгоритме.
```

**Review by saionaro — APPROVED** [review]
@scffs спасибо за дополнения к переводу!

File: files/ru/web/javascript/equality_comparisons_and_sameness/index.md [hunk 6]
```diff
@@ -186,23 +186,23 @@ function attemptMutation(v) {
...
+Аналогично равенству одинаковых величин, но +0 и -0 считаются равными.
 
-Same-value-zero equality is not exposed as a JavaScript API, but can be implemented with custom code:
+Равенство одинаковых величин и нулей не предоставляется как JavaScript API, но может быть реализовано с помощью пользовательского кода:
 
 ```js
 function sameValueZero(x, y) {
   if (typeof x === "number" && typeof y === "number") {
-    // x and y are equal (may be -0 and 0) or they are both NaN
+   // x и y равны (могут быть -0 и 0) или они оба равны NaN
```
Comment (Saionaro) [inline files/ru/web/javascript/equality_comparisons_and_sameness/index.md:198]:
```suggestion
    // x и y равны (могут быть -0 и 0) или они оба равны NaN
```

**Comment by scffs** [comment]
thx for review

**Pull request merged**